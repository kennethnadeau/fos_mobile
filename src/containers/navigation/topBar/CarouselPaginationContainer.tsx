import React from "react";
import { ScreenFC } from "react-native-navigation-register-screens";
import { SCREENS } from "@fos/constants";
import { useSelector } from "react-redux";
import { selectPagination } from "@fos/redux/selectors/navigationSelectors";
import { CarouselPagination } from "@fos/components/navigation/topBar";

const CarouselPaginationContainer: ScreenFC = () => {
  const pagination = useSelector(selectPagination);

  return <CarouselPagination {...pagination} />;
};

CarouselPaginationContainer.screenName = SCREENS.COMPONENTS.TOP_BAR.PAGINATION;

export default CarouselPaginationContainer;
