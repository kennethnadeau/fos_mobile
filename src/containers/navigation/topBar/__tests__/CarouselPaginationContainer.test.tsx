import React from "react";
import { render, RenderAPI } from "@fos/utils/testing";
import CarouselPaginationContainer from "../CarouselPaginationContainer";

describe("<CarouselPaginationContainer />", () => {
  let carouselPaginationContainer: RenderAPI;

  beforeEach(() => {
    carouselPaginationContainer = render(
      <CarouselPaginationContainer componentId="id" />,
    );
  });

  // eslint-disable-next-line jest/no-disabled-tests
  it.skip("should match snapshot", () =>
    expect(carouselPaginationContainer.toJSON()).toMatchSnapshot());
});
