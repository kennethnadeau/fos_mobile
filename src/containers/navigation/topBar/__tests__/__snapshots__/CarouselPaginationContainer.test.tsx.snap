// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselPaginationContainer /> should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="Carousel pagination"
    accessible={true}
  >
    <View
      accessible={false}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 20,
            "paddingVertical": 30,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "paddingVertical": 0,
          },
        ]
      }
    >
      <View
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(126, 131, 149, 1)",
              "borderRadius": 3.5,
              "height": 7,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
              "width": 7,
            }
          }
        />
      </View>
      <View
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(126, 131, 149, 1)",
              "borderRadius": 3.5,
              "height": 7,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
              "width": 7,
            }
          }
        />
      </View>
      <View
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(126, 131, 149, 1)",
              "borderRadius": 3.5,
              "height": 7,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
              "width": 7,
            }
          }
        />
      </View>
      <View
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(126, 131, 149, 1)",
              "borderRadius": 3.5,
              "height": 7,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
              "width": 7,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
